Lecture2/src/task5/Task5App.java

```diff
@@ -0,0 +1,103 @@
+package task5;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Task5App {
+
+    private final static char SYMBOL = '*';
+    private static Scanner scanner = new Scanner(System.in);
+    private static List<String> variants = Arrays.asList("Vitaly Zenkou", "Vlad Bankov", "Vitaly Plisko");
+    private static List<String> guessesCharacter = new ArrayList<>();
+    private static String hiddenName;
+    private static StringBuffer result = new StringBuffer();
+    private static boolean isWin;
+
+    public static void main(String[] args) {
+        chooseHiddenName();
+        do {
+            System.out.println("The hidden name is " + result.toString());
+            String str = readString();
+            compare(str);
+            isWin = isWin();
+        } while (!isWin);
+        System.out.println("You win! The hidden name is " + hiddenName);
+    }
+
+    private static void chooseHiddenName() {
```

Comment:
common suggestion - static var which changes in different methods - bad practice 
make chooseHiddenName returns String  - and work with this var in main method (pass it in param to other methods)

Reply:
fix